apps/frontend/src/components/ViewFormsList/ViewFormsList.tsx

```diff
@@ -14,18 +14,72 @@ import {
   Thead,
   Tr,
   Link,
+  Input,
 } from '@chakra-ui/react';
 import { getAllForms } from '../../utils/sendRequest';
 import { FormData } from '../../types/formData';
+import { SortOptions } from '../../enums/SortOrder';
 
 export default function ViewFormsList() {
-  const [forms, setForms] = useState<FormData[] | null>(null);
+  const [forms, setForms] = useState<FormData[]>([]);
+  const [allForms, setAllForms] = useState<FormData[]>([]); // this is used to get all forms again after removing a filter/search term
+  const [sortBy, setSortBy] = useState<SortOptions>(SortOptions.NAME);
+  const [searchTerm, setSearchTerm] = useState<string>('');
 
   const getForms = async () => {
     const allForms = await getAllForms();
     setForms(allForms);
+    setAllForms(allForms);
   };
 
+  // sort forms
+  useMemo(() => {
+    if (sortBy === SortOptions.NAME) {
+      forms.sort((a, b) =>
+        a.guardianForm.childsName.localeCompare(b.guardianForm.childsName)
+      );
+    } else if (sortBy === SortOptions.LASTUPDATED) {
+      forms.sort(
+        (a, b) =>
+          (b.adminNotes.length > 0
+            ? new Date(b.adminNotes[0].updatedAt).getTime()
+            : new Date(b.guardianForm.date).getTime()) -
+          (a.adminNotes.length > 0
+            ? new Date(a.adminNotes[0].updatedAt).getTime()
+            : new Date(a.guardianForm.date).getTime())
+      );
+    }
+  }, [sortBy, forms]);
+
+  // filter forms by search term
+  useMemo(() => {
+    if (searchTerm.length > 0) {
+      setForms(
+        forms.filter((form) => {
+          const formValues = Object.values({
+            ...form.guardianForm,
+            ...form.medicalForm,
+            ...form.guardianForm.address,
+          });
+          console.log(formValues);
```

Comment:
Did you mean to leave these console.logs in? If so, should probably add a comment  to them to explain what they are

Reply:
oop my bad thanks!!